odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,124 @@ def _validate_module_views(self, module):
+                    # should not happend
+
+    def _get_faulty_views(self,parent_view=None):
+        """ Returns the faulty views
+            To determine the faulty views, we travel the tree recursively.
+            Disable all children, then enable one by one. If there is an issue.
+            The activated node or at least one of its children are faulty.
+            Disable faulty node and continue recursion.
+            This function is never called from odoo code but allows to be called from a server action.
+            Please don't remove
+    usage:
+        view1._get_faulty_views()
+        """
+        self.ensure_one()
+        disabled_views = self.env['ir.ui.view']
+        parent_view = parent_view or self
+        if parent_view==self:
+            if self._is_valid():
+                return disabled_views
+            self._cr.execute('SAVEPOINT test_fix_views')
+            temp_disabled_views=self._deactivate_lvl1_children() #ordered by priority ( by default)
+            if self._is_valid():
+                if len(self.inherit_children_ids) or len(temp_disabled_views)>1:
+                    for child_view in (temp_disabled_views):
+                        for v in child_view._get_faulty_views(self):
+                            disabled_views|=v       
+                else:
+                    self.active=False
+                    disabled_views+=self
+            else:
+                self.active=False
+                disabled_views+=self
+            self._cr.execute('ROLLBACK TO test_fix_views')
+            self._cr.execute('RELEASE test_fix_views')
+            return disabled_views
+        else:
+            assert parent_view._is_valid()
+            self.active=True
+            if parent_view._is_valid():
+                return disabled_views
+            if len(self.inherit_children_ids):
+                temp_disabled_views=self._deactivate_lvl1_children()
+                if parent_view._is_valid():
+                    # problem is in one of the children
+                    for child_view in (temp_disabled_views):
+                        for v in child_view._get_faulty_views(parent_view):
+                            disabled_views|=v       
+                    return disabled_views
+                else:
+                    self.active=False
+                    disabled_views+=self
+                    return disabled_views
+            else:
+                self.active=False
+                disabled_views+=self
+                return disabled_views
+
+    def _deactivate_lvl1_children(self):
+        """deactivate and return views that has been deactivated taking care of view priority"""
+        temp_disabled_views= self.inherit_children_ids
+        temp_disabled_views.update({'active': False})
+        return temp_disabled_views
+
+    def _is_valid(self):
```

Comment:
something i did not expect you to know, but a nice addition nonetheless: when using this function, it will potentially write a lot of errors in the logs for something that's just exploring a tree:
```
2019-04-09 08:33:18,103 29617 INFO broken-views werkzeug: 127.0.0.1 - - [09/Apr/2019 08:33:18] "POST /web/action/load HTTP/1.1" 200 - 14 0.009 0.011
2019-04-09 08:33:18,162 29617 INFO broken-views odoo.addons.base.models.ir_ui_view: Element '<xpath expr="//div[@id='d2bo']">' cannot be located in parent view

Error context:
View `Test DBO 2`
[view_id: 453, xml_id: n/a, model: crm.lead, parent_id: 452] 
2019-04-09 08:33:18,311 29617 INFO broken-views odoo.addons.base.models.ir_ui_view: Element '<xpath expr="//div[@id='d2bo']">' cannot be located in parent view

Error context:
View `Test DBO 2`
[view_id: 453, xml_id: n/a, model: crm.lead, parent_id: 452] 
2019-04-09 08:33:18,458 29617 INFO broken-views odoo.addons.base.models.ir_ui_view: Element '<xpath expr="//div[@id='d2bo']">' cannot be located in parent view

Error context:
View `Test DBO 2`
[view_id: 453, xml_id: n/a, model: crm.lead, parent_id: 452] 
2019-04-09 08:33:18,538 29617 INFO broken-views odoo.addons.base.models.ir_ui_view: View 428 can be fixed by disabling/deleting following views: ir.ui.view(453,) 
```

the log can be disabled by using the `mute_logger` decorator from `odoo/tools/misc.py` like so:
``` python
    @mute_logger('odoo.addons.base.models.ir_ui_view')
    def _is_valid(self):
        # do stuff
```
(+ a from `odoo.tools import mute_logger` at the beginning of the file)

Reply:
can be used in tests as well

Reply:
done thank you very much